File: camps/summer_school_Kremenchuk_2016/kkzp/D.cpp
#include<iostream>
#include<fstream>

using namespace std;

long long getCountOfPowPair(long long summ)
{
  long long ans = 0;
  for(long long i = 1; i*i<summ; i++)
    ans+=summ-i*i;
  return ans;
}

int solve(long long n)
{
  long long ans = 0;
  for(long long i = 1; i*i*i<n; i++)
  {
    ans+=getCountOfPowPair(n-i*i*i);
  }
  return ans;
}
int main()
{
  ofstream cout("D_outputA.txt");
  for(int i = 3; i<1000; i++)
    cout << i << ' ' << solve(i) << endl;

  return 0;
}

File: camps/summer_school_Kremenchuk_2016/kkzp/D.py
def a(b):
    ans = 0
    i = 1
    while(i*i<b):
        ans+=b-i*i
        i+=1
    return ans
ans = 0
i = 1
n = 1000000000000
while(i*i*i<n):
    if(i%100==0):
        print(n-i*i*i)
    ans+=a(n-i*i*i)
    i+=1
print(ans)
	
